File: api-docs/packages/hnb-venus-api-docs/typescript/examples/smoke.ts
import { HnbVenusApiDocsClient } from "../src/index.js";

/**
 * Live smoke — polite delay on. Run: npm run smoke
 * May fail if upstream is down / blocks this environment.
 */
async function main() {
  const client = new HnbVenusApiDocsClient({ defaultDelayMs: 1000 });
  console.log("smoke", HnbVenusApiDocsClient.slug, "->", "getExchangeRatesContentsWeb");
  const data = await client.getExchangeRatesContentsWeb();
  const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
  console.log("ok", preview);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: api-docs/packages/hnb-venus-api-docs/python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from hnb_venus_api_docs import HnbVenusApiDocsClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with HnbVenusApiDocsClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "get_exchange_rates_contents_web")
    data = client.get_exchange_rates_contents_web()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)
